lib/repl.js

```diff
@@ -778,7 +785,14 @@ function REPLServer(prompt,
       self.lines.level = [];
       self.displayPrompt();
     }
-  });
+  };
+  kListeningREPLs.add(self);
+
+  if (!kHasSetUncaughtListener) {
+    kAsyncHook.enable();
+    process.on('uncaughtException', handleUncaughtException);
```

Comment:
This is still leaking things in the process object, and domain is a much safer way to do so anyway.

Reply:
> and domain is a much safer way to do so anyway.

I disagree. This is different than the domain. The domain bypasses the listener and inserts a callback before these listeners can even be called. This merely adds a listener, and later removes it. How is that "leaking"? The attachment to the process object is controlled.

Reply:
What is happening is that the specified `uncaughtException` handler will intercepts errors coming from other part of the applications, and vice versa errors coming from the REPL could be caught by `uncaughtException` added by the user.

Reply:
To be extra careful, I've added tests to ensure this behavior is accurate. See <https://github.com/nodejs/node/pull/55204/files#diff-ca6f6e3faa6e2dc3fba74eabfc212f36fa2f84b7a17e3f73662d24b678d18f73>